Data Apps CRUD (#60569)
metabase/metabase: src/metabase/data_app/CLAUDE.md

- Validation occurs before insert/update operations
- Retracted flag allows marking releases as unavailable when needed

## API Endpoints

The data apps feature exposes two distinct sets of API endpoints:

### Admin/Management APIs (`/api/data-app/`)

> Located in `api/data-apps.clj`, these endpoints are used for creating, updating, and managing data apps:

Reviewer: ```suggestion
Located in `api/data-apps.clj`, these endpoints support the manipulation of Data Apps by Creators.
```

> These endpoints require appropriate permissions (create/update/delete checks) and are intended for app authors and administrators.

Reviewer: Oh, I didn't know about the generic api `update-check` etc methods, so I guess we can mention that they *should* enforce the perms.

For this 6 week iteration I think the the permission model will be as simple as only letting super users create and edit apps.

### Consumer APIs (`/api/app/`)

> Located in `api/app.clj`, these endpoints are used by end-users to interact with published data apps:

Reviewer: ```suggestion
Located in `api/app.clj`, these endpoints are used to run Data Apps for Consumers.
```
